testsuite/emscripten/quiet_emcc_info.py

```diff
@@ -0,0 +1,33 @@
+"""
+Prevent emcc from printing info level logging
+
+dejagnu will fail tests because of these log statements and the messages depend
+on emscripten cache state so xfailing the right tests is very hard.
+"""
+
+from pathlib import Path
+from shutil import which
+from sys import exit
+from textwrap import dedent
+
+EMCC_PATH = Path(which("emcc") + ".py")
+
+emcc_text = EMCC_PATH.read_text()
+if "# quiet_emcc_info patch" in emcc_text:
+    exit(0)
+
+emcc_lines = emcc_text.splitlines()
+emcc_lines.insert(
+    1,
+    dedent(
+        """
+        # quiet_emcc_info patch
+        import logging
+        for name in ["cache", "system_libs", "shared"]:
+            logger = logging.getLogger(name)
+            logger.setLevel(logging.WARN)
```

Comment:
@sbc100 for now it seems like this is a reasonable workaround for #18607.

Reply:
Oh yeah I guess so.  Are you actually patching emcc.py here?   Long term seems risky but short term I guess YOLO.

Reply:
YOLO indeed =)

This was the first method I tried that actually worked. I was unable to convince the test system dejagnu that the tests should pass despite these `INFO` messages, even though that should be a reasonable thing to ask it to do. Maybe @atgreen can tell me how to do that.

Another approach would be to replace `emmake` with my own wrapper script and use a compiler wrapper. But that would require me to duplicate the logic in `get_building_env`.

Reply:
If you are trying to filter out some output for dejagnu, check out libffi-dg-prune in testsuite/lib/libffi.exp.  Will that work?

Reply:
Yeah this works great. I should have just asked! Will make a PR to fix this.